### PR TITLE
feat: Added custom location overwrite

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

- Added custom location overwrite
- This is important if you want to rerun / test the same resource deployment more than once (e.g. because you disabled the removal the first time around) and don't want the location rotation to complicate the effort

image

image

## Pipeline Reference

| Pipeline |
| -------- |
|    [avm.res.key-vault.vault](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.key-vault.vault.yml)      |

image

## Type of Change

- [x] Update to CI Environment or utlities (Non-module effecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

### CODE REVIEW

**Review by ChrisSidebotham — DISMISSED** [review]
LGTM

**Comment by AlexanderSehr** [comment]
Will wait for @eriqua to also take a look / give feedback before I merge. Thanks @ChrisSidebotham 💪

**Comment by ChrisSidebotham** [comment]
@AlexanderSehr - Can we merge this soon before we risk it getting out of date?

**Comment by AlexanderSehr** [comment]
> @AlexanderSehr - Can we merge this soon before we risk it getting out of date?

Following up on my previous comment, I still think @eriqua would want a look as it touches the CI. Trying to be considerate ;)

File: .github/workflows/avm.ptn.authorization.policy-assignment.yml
Comment (eriqua) [inline .github/workflows/avm.ptn.authorization.policy-assignment.yml:23]: Hello there 👋 
Let's say this is a way to overcome excluded locations, but I don't think it would harm considering it can only work when manually triggered (I.e., when publishing is disabled)

Fine for me, but waited for all CARML modules to be merged, so I think latest published modules need to be added.

Comment (eriqua) [reply .github/workflows/avm.ptn.authorization.policy-assignment.yml:23]: Also, it would probably be useful to add an example in the field, to show which convention to use (e.g. West Europe or westeurope)

Comment (AlexanderSehr) [reply .github/workflows/avm.ptn.authorization.policy-assignment.yml:23]: Updated to 
image

File: .github/workflows/avm.template.module.yml
Comment (eriqua) [inline .github/workflows/avm.template.module.yml:22]: Since part of `workflowInput` I think it's not needed here

Comment (AlexanderSehr) [reply .github/workflows/avm.template.module.yml]: fixed

File: .github/workflows/avm.ptn.authorization.policy-assignment.yml
Comment (AlexanderSehr) [reply .github/workflows/avm.ptn.authorization.policy-assignment.yml:23]: I think this may be due to other PRs having merged again in the meantime. As @ChrisSidebotham pointed out, this PR runs high risk of running stale fast 😄 I'm trying to find out which are missing. Looking at all workflows it should be 132, not 130.

Comment (AlexanderSehr) [reply .github/workflows/avm.ptn.authorization.policy-assignment.yml:23]: Should be fixed now

Comment (AlexanderSehr) [reply .github/workflows/avm.ptn.authorization.policy-assignment.yml:23]: @eriqua, this is neither the time nor place to let people know that I did not pay attention in math class.

Comment (AlexanderSehr) [reply .github/workflows/avm.ptn.authorization.policy-assignment.yml:23]: According to a friend, the numbers should add up now

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**
